agreement_legal_sale/models/sale_order.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (C) 2019 - TODAY, Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    agreement_template_id = fields.Many2one(
+        "agreement", string="Agreement Template", domain="[('is_template', '=', True)]"
+    )
+
+    def _action_confirm(self):
+        res = super(SaleOrder, self)._action_confirm()
```

Comment:
You can adopt short expression 
```suggestion
        res = super()._action_confirm()
```